### PR TITLE
Reimplement the QuarkusJTAPlatform to avoid leaking the ORM registry

### LINKED ISSUE #12976: Hibernate ORM's JTAPlatform implementation is leaking a reference to its registry
**Describe the bug**
The `QuarkusJtaPlatform` implementation is extending a base class from upstream ORM which maks it a bad choice for also being used as a static singleton.
In particular this leads the classloader to hold on to a reference of the ORM `ServiceRegistry`, in turn leaking a significant amount of memory.

We can easily fix it to be a real stateless JtaPlatform, as we don't need the registry.

### PR DESCRIPTION
**Pull request by Sanne**
Fixes #12976

### CODE REVIEW

File: extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/customized/QuarkusJtaPlatform.java [hunk 1]
```diff
@@ -1,26 +1,72 @@
...
     private QuarkusJtaPlatform() {
         //nothing
     }
 
     @Override
-    protected TransactionManager locateTransactionManager() {
-        return com.arjuna.ats.jta.TransactionManager.transactionManager();
+    public TransactionManager retrieveTransactionManager() {
+        TransactionManager transactionManager = this.transactionManager;
+        if (transactionManager == null) {
```
Comment (gsmet) [inline extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/customized/QuarkusJtaPlatform.java:30]: Looks like some double checked locking except I don't see the `synchronized` part.

Is it safe?

File: extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/customized/QuarkusJtaPlatform.java [hunk 2]
```diff
@@ -1,26 +1,72 @@
...
+        return retrieveTransactionManager();
+    }
+
+    @Override
+    public UserTransaction retrieveUserTransaction() {
+        UserTransaction userTransaction = this.userTransaction;
+        if (this.userTransaction == null) {
+            userTransaction = com.arjuna.ats.jta.UserTransaction.userTransaction();
+            this.userTransaction = userTransaction;
+        }
```
Comment (gsmet) [inline extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/customized/QuarkusJtaPlatform.java:48]: Same question here.

**Review by gsmet — CHANGES_REQUESTED** [review]
Added some questions.

File: extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/customized/QuarkusJtaPlatform.java (hunk 1 above)
Comment (Sanne) [reply extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/customized/QuarkusJtaPlatform.java:30]: it's safe as the method we invoke is `synchronized` and returns a singleton.

File: extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/customized/QuarkusJtaPlatform.java (hunk 2 above)
Comment (Sanne) [reply extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/customized/QuarkusJtaPlatform.java:48]: same :)

**Review by gsmet — APPROVED** [review]

**Pull request merged**